src/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -541,6 +541,35 @@ public static void DeserializeIntoJsonElementProperty()
             Assert.Equal(JsonValueKind.Object, obj.MyOverflow["MyOverflow"].ValueKind);
         }
 
+        [Fact]
+        public static void SerializerOutputRoundtripsWhenEscaping()
+        {
+            string jsonString = "{\"\u6C49\u5B57\":\"abc\",\"Class\":{\"\u6F22\u5B57\":\"xyz\"},\"\u62DC\u6258\":{\"\u62DC\u6258\u62DC\u6258\":1}}";
+
+            ClassWithEscapedProperty input = JsonSerializer.Deserialize<ClassWithEscapedProperty>(jsonString);
+
+            Assert.Equal("abc", input.汉字);
```

Comment:
What can we do about property name accesses like these? This is no longer ASCII.

Reply:
You seem to have already made the member declaration be \u6C49\u5B57, theoretically you can do the same here...

```suggestion
            Assert.Equal("abc", input.\u6C49\u5B57);
```

Reply:
That's hilarious. I should have tried that, but was relying on auto-complete, which obviously shows the un-escaped name.